receiver/mongodbatlasreceiver/logs.go

```diff
@@ -0,0 +1,306 @@
+// limitations under the License.
+
+package mongodbatlasreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbatlasreceiver"
+
+import (
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"net"
+	"strings"
+	"sync"
+	"time"
+
+	"go.mongodb.org/atlas/mongodbatlas"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbatlasreceiver/internal"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbatlasreceiver/internal/model"
+)
+
+type logsReceiver struct {
+	log         *zap.Logger
+	cfg         *Config
+	client      *internal.MongoDBAtlasClient
+	consumer    consumer.Logs
+	stopperChan chan struct{}
+	wg          sync.WaitGroup
+}
+
+type ProjectContext struct {
+	Project   mongodbatlas.Project
+	orgName   string
+	projectID string
+	start     time.Time
+	end       time.Time
+}
+
+// MongoDB Atlas Documentation reccommends a polling interval of 5  minutes: https://www.mongodb.com/docs/atlas/reference/api/logs/#logs
+const collectionInterval = time.Minute * 5
+
+func newMongoDBAtlasLogsReceiver(settings component.ReceiverCreateSettings, cfg *Config, consumer consumer.Logs) (*logsReceiver, error) {
+	client, err := internal.NewMongoDBAtlasClient(cfg.PublicKey, cfg.PrivateKey, cfg.RetrySettings, settings.Logger)
+	if err != nil {
+		return nil, err
+	}
+	return &logsReceiver{
+		log:         settings.Logger,
+		cfg:         cfg,
+		client:      client,
+		stopperChan: make(chan struct{}),
+		consumer:    consumer}, nil
+}
+
+// Log receiver logic
+func (s *logsReceiver) Start(ctx context.Context, host component.Host) error {
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+		start := time.Now()
+		for {
+			// collection interval loop,
+			select {
+			case <-ctx.Done():
+				return
+			case <-s.stopperChan:
+				return
+			case <-time.After(collectionInterval):
+			}
+			start = s.collect(ctx, start)
+		}
+	}()
+	return nil
+}
+
+func (s *logsReceiver) Shutdown(ctx context.Context) error {
+	close(s.stopperChan)
+	s.wg.Wait()
+	return s.client.Shutdown()
+}
+
+// parseHostNames parses out the hostname from the specified cluster host
+func parseHostNames(s string, logger *zap.Logger) []string {
+	var hostnames []string
+
+	if s == "" {
+		return []string{}
+	}
+
+	for _, t := range strings.Split(s, ",") {
+		// separate hostname from scheme and port
+		host, _, err := net.SplitHostPort(strings.TrimPrefix(t, "mongodb://"))
+		if err != nil {
+			logger.Error("Could not parse out hostname: " + host)
+			continue
+		}
+		hostnames = append(hostnames, host)
+	}
+
+	return hostnames
+}
+
+// collect spins off functionality of the receiver from the Start function
+func (s *logsReceiver) collect(ctx context.Context, start time.Time) time.Time {
+	pc := ProjectContext{start: start, end: time.Now()}
```

Comment:
Now that we are not updating `end` for each project, it looks like we are consistently using the same `start` and `end` for all projects during each collection cycle. This is great. Now that these are basically singleton values, we might as well just store them on the receiver. Passing them around will be unnecessary. The values should only be initialized in the `Start` function and updated at the end of the `collect` function.